src/share/poudriere/jail.sh

```diff
@@ -482,6 +483,10 @@ setup_src_conf() {
 	[ -f ${POUDRIERED}/${JAILNAME}-${src}.conf ] && \
 	    cat ${POUDRIERED}/${JAILNAME}-${src}.conf >> \
 	    ${JAILMNT}/etc/${src}.conf
+
+	[ -n "${PKG_REPO_SIGNING_KEY}" ] && [ "${src}" = "src" ] && \
+	    echo "PKG_REPO_SIGNING_KEY=${PKG_REPO_SIGNING_KEY}" >> \
+	    ${JAILMNT}/etc/${src}.conf
```

Comment:
It might be cleaner to only set PKG_REPO_SIGNING_KEY where it is needed, at the `make update-packages` call and then only in that call's environment. Thus keeping all of the logic for packages in the same place.